test/replication/qsync_advanced.result

```diff
@@ -746,6 +786,9 @@ box.space.sync:select{} -- 1, 2, 3
  |   - [2]
  |   - [3]
  | ...
+test_run:wait_lsn('replica', 'default')
```

Comment:
There is a typo a couple of lines above:
```
-- Enable synchronous mode.
disable_sync_mode()
 | ---
 | ...
```

This is why this place flaked without wait_lsn. It tries to test turning an async space to sync, but did nothing instead.
Besides, the writes with BROKEN_QUORUM shouldn't succeed.

Reply:
So to address this I introduced the `enable_sync_mode()` and changed the logic of the "3" insertion.